onnx/shape_inference/implementation.cc

```diff
@@ -84,6 +84,10 @@ void checkShapesAndTypes(
     const TypeProto& existingType) {
   const auto inferredTypeCase = inferredType.value_case();
   const auto existingTypeCase = existingType.value_case();
+  if (existingTypeCase == TypeProto::ValueCase::VALUE_NOT_SET) {
```

Comment:
Please add "|| inferredTypeCase == TypeProto::ValueCase::VALUE_NOT_SET" to the if-condition. It will make it more robust, I think.